pubtools/pulplib/_impl/client/client.py

```diff
@@ -302,3 +326,40 @@ def _delete_resource(self, resource_type, resource_id):
 
         LOG.debug("Queuing request to DELETE %s", url)
         return self._task_executor.submit(self._do_request, method="DELETE", url=url)
+
+    def _request_upload(self):
+        url = os.path.join(self._url, "pulp/api/v2/content/uploads/")
+
+        LOG.debug("Requesting upload id")
+        return self._request_executor.submit(self._do_request, method="POST", url=url)
+
+    def _do_upload(self, data, upload_id, offset):
+        url = os.path.join(
+            self._url, "pulp/api/v2/contents/%s/%s" % (upload_id, offset)
```

Comment:
This seems like the wrong URL.  See the docs at: https://docs.pulpproject.org/dev-guide/integration/rest-api/content/upload.html#upload-bits

```suggestion
            self._url, "pulp/api/v2/content/uploads/%s/%s" % (upload_id, offset)
```